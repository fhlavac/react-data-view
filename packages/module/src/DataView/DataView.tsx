import React from 'react';
import { Stack, StackItem } from '@patternfly/react-core';
import { DataViewSelection, InternalContextProvider } from '../InternalContext';

export const DataViewState = {
  empty: 'empty',
  loading: 'loading',
  error: 'error'
} as const;

export type DataViewState = typeof DataViewState[keyof typeof DataViewState];

export interface DataViewProps {
  /** Content rendered inside the data view */
  children: React.ReactNode;
  /** Custom OUIA ID */
  ouiaId?: string;
  /** Selection context configuration */
  selection?: DataViewSelection;
  /** Currently active state */
  activeState?: DataViewState | string;
}

export type DataViewImpementationProps = Omit<DataViewProps, 'onSelect' | 'isItemSelected' | 'isItemSelectDisabled'>;

const DataViewImplementation: React.FC<DataViewImpementationProps> = ({
  children, ouiaId = 'DataView', ...props
}: DataViewImpementationProps) => (
  <Stack data-ouia-component-id={`${ouiaId}-stack`} {...props}>
    {React.Children.map(children, (child, index) => (
      <StackItem data-ouia-component-id={`${ouiaId}-stack-item-${index}`}>
        {child}
      </StackItem>
    ))} 
  </Stack>
)

export const DataView: React.FC<DataViewProps> = ({ children, selection, activeState, ...props }: DataViewProps) => (
  <InternalContextProvider selection={selection} activeState={activeState} >
    <DataViewImplementation {...props}>{children}</DataViewImplementation>
  </InternalContextProvider>
);

export default DataView;
