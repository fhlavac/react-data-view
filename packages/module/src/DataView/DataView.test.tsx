import React from 'react';
import { render } from '@testing-library/react';
import DataView from './DataView';

const layoutItemStyling = {
  width: '100%',
  height: '5rem',
  padding: 'var(--pf-v5-global--spacer--md)',
  borderStyle: 'dashed',
  borderWidth: '2px',
};

describe('DataView component', () => {
  test('should render correctly', () => {
    expect(render(
      <DataView>
        <div style={layoutItemStyling}>Header</div>
        <div style={{ ...layoutItemStyling, borderTopWidth: 0, borderBottomWidth: 0 }}>Data representation</div>
        <div style={layoutItemStyling}>Footer</div>
      </DataView>)).toMatchSnapshot();
  });
});