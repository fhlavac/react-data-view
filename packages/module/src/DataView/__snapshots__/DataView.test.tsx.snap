// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataView component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-v5-l-stack"
        data-ouia-component-id="DataView-stack"
      >
        <div
          class="pf-v5-l-stack__item"
          data-ouia-component-id="DataView-stack-item-0"
        >
          <div
            style="width: 100%; height: 5rem; border-style: dashed; border-width: 2px;"
          >
            Header
          </div>
        </div>
        <div
          class="pf-v5-l-stack__item"
          data-ouia-component-id="DataView-stack-item-1"
        >
          <div
            style="width: 100%; height: 5rem; border-style: dashed; border-width: 2px; border-top-width: 0; border-bottom-width: 0;"
          >
            Data representation
          </div>
        </div>
        <div
          class="pf-v5-l-stack__item"
          data-ouia-component-id="DataView-stack-item-2"
        >
          <div
            style="width: 100%; height: 5rem; border-style: dashed; border-width: 2px;"
          >
            Footer
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-v5-l-stack"
      data-ouia-component-id="DataView-stack"
    >
      <div
        class="pf-v5-l-stack__item"
        data-ouia-component-id="DataView-stack-item-0"
      >
        <div
          style="width: 100%; height: 5rem; border-style: dashed; border-width: 2px;"
        >
          Header
        </div>
      </div>
      <div
        class="pf-v5-l-stack__item"
        data-ouia-component-id="DataView-stack-item-1"
      >
        <div
          style="width: 100%; height: 5rem; border-style: dashed; border-width: 2px; border-top-width: 0; border-bottom-width: 0;"
        >
          Data representation
        </div>
      </div>
      <div
        class="pf-v5-l-stack__item"
        data-ouia-component-id="DataView-stack-item-2"
      >
        <div
          style="width: 100%; height: 5rem; border-style: dashed; border-width: 2px;"
        >
          Footer
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
